### PR TITLE
Add --include-deps to push command

### PR DESCRIPTION
**Pull request by gferon**
**What I did**

Add `--include-deps` to the `push` sub-command, similar to what `pull` can do. I believe the default behavior is a little bit weird since you can't explicitly push the built image of a service that has declared dependencies. That being said, this change will break backwards compatibility, so I feel like we could

_Cute animal tax_ - a photo of my containerized cat:

PXL_20201120_110811709 PORTRAIT(2)

### CODE REVIEW

**Comment by ndeloof** [comment]
While I don't expect many users to rely on this, I'm a bit concerned about another backward compatibility break.
Maybe better introduce `--no-deps` to mimic same option used by `up`?
Also, to avoid code duplication, could define a func to disable all services but selected ones

**Comment by gferon** [comment]
> While I don't expect many users to rely on this, I'm a bit concerned about another backward compatibility break.

Fair point, I can definitely move it to a `--no-deps` flag instead, but this would be inconsistent with `docker-compose pull`. What about setting the value of this flag to `true` instead and let users pass `--include-deps=false` instead?

> Also, to avoid code duplication, could define a func

Sure, can you point me to a good place to do this? I'm not familiar with this codebase to select a proper location myself.

**Comment by ndeloof** [comment]
you can just extract https://github.com/docker/compose/blob/v2/cmd/compose/up.go#L57-L66 into a func inside up.go and call it from both command.

I didn't noticed `--include-deps` flag existed for pull, makes more sense now. Both should be aligned. Minor compatibility break probably is acceptable from this point of view.

**Comment by ndeloof** [comment]
also need to regenerate docs to reflect new flag using `make docs`

**Comment by gferon** [comment]
@ndeloof not sure why you meant `up.go`, I extracted the function in `pull.go` - if you really wanted it to be in `up.go` I can change it.

Otherwise, this should be ready.

File: cmd/compose/pull.go [hunk 1]
```diff
@@ -67,23 +68,29 @@ func pullCommand(p *projectOptions, backend api.Service) *cobra.Command {
 	return cmd
 }
 
+func FilterServices(project *types.Project, services []string) error {
```
Comment (ndeloof) [inline cmd/compose/pull.go:71]: Don't need to be exporte. Also name is bit confusing. Maybe `withSelectedServicesOnly` or something comparable?

Comment (gferon) [reply cmd/compose/pull.go]: Done!

**Review by ndeloof — APPROVED** [review]

**Pull request merged**